packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -372,30 +380,65 @@ class ClearSelectionEvent extends SelectionEvent {
   const ClearSelectionEvent(): super._(SelectionEventType.clear);
 }
 
+/// An abstract base class for selection events that select a boundary at a location.
+///
+/// See also:
+/// * [SelectWordSelectionEvent], for an event that selects the word at the location.
+/// * [SelectParagraphSelectionEvent], for an event that selects the paragraph at the location.
+/// * [SelectLineSelectionEvent], for an event that selects the line at the location.
+/// * [SelectionEventType], for determining the subclass types.
+abstract class SelectBoundarySelectionEvent extends SelectionEvent {
+  const SelectBoundarySelectionEvent._(this.globalPosition, SelectionEventType type) : super._(type);
+
+  /// The position in global coordinates to select the boundary at.
+  final Offset globalPosition;
+}
+
+/// An abstract base class for selection events that select a boundary that
+/// may span multiple [Selectable]s.
+///
+/// See also:
+/// * [SelectParagraphSelectionEvent], for an event that selects the paragraph at the location.
+/// * [SelectLineSelectionEvent], for an event that selects the line at the location.
+/// * [SelectionEventType], for determining the subclass types.
+abstract class SelectMultiSelectableBoundarySelectionEvent extends SelectBoundarySelectionEvent {
+  const SelectMultiSelectableBoundarySelectionEvent._(this.absorb, Offset globalPosition, SelectionEventType type) : super._(globalPosition, type);
+
+  /// Whether the [Selectable] receiving the event should be completely
+  /// absorbed into a boundary.
+  ///
+  /// For example, when [absorb] is set to true, and a [Selectable] that is inline
+  /// within a boundary receives this event, it should select all of its content,
+  /// and continue searching for the end of the boundary.
+  ///
+  /// When [absorb] is set to false, and a [Selectable] that is inline with a boundary
+  /// receives this event, it should find the end of the boundary relative to itself
+  /// and stop searching.
+  final bool absorb;
```

Comment:
based on the doc and the code, can we replace case with absorb = true with a select all event?

Reply:
Answered in: https://github.com/flutter/flutter/pull/153514#discussion_r1823203271